virttest/qemu_devices/qdevices.py

```diff
@@ -172,6 +172,8 @@ def __eq__(self, dev2, dynamic=True):
         """ :return: True when devs are similar, False when different. """
         if not isinstance(dev2, QBaseDevice):
             return False
+        if isinstance(dev2, QDaemonDev):
+            return dev2.__eq__(self)
```

Comment:
Okay, I'm just curious about why it is handled by the base class instead of the derived class, is there some reason for that?

Reply:
> Okay, I'm just curious about why it is handled by the base class instead of the derived class, is there some reason for that?

Yes, I also have such confusing. From my ideal design, I would like to handle it in the class QDaemonDev, but I can not
figure out the better way to override the following `__eq__` implementation of comparing `dev2` from the QBaseDevice:
```
    def __eq__(self, dev2, dynamic=True):
        ...
        check_attrs = ['cmdline_nd', 'hotplug_hmp_nd', 'hotplug_qmp_nd']
        try:
            for check_attr in check_attrs:
                try:
                    _ = getattr(self, check_attr)()
                except (DeviceError, NotImplementedError, AttributeError):
                    ...
                else:
                    if _ != getattr(dev2, check_attr)():
                        return False
        ...
```
For the sentence `if _ != getattr(dev2, check_attr)():` however it can call the `dev2.cmdline_nd()` so I can not override it.
Maybe we consider to re-design the implementation of `__eq__`.
And looks like my way just a workaround. Do you have any better solution?
Thanks.